chap1/main.go

```diff
@@ -1,9 +1,107 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/NozomiSugiyama/study-golang/chap1/controller"
+	"github.com/NozomiSugiyama/study-golang/chap1/model"
 )
 
 func main() {
-	fmt.Println("Hello Chap1")
+	http.HandleFunc("/users", usersHandler)
+
+	http.ListenAndServe(":8080", nil)
+}
+
+func usersHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
+		users, err := controller.ListUsers()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		switch r.URL.Query().Get("format") {
+		case "html":
+			w.Header().Set("Content-Type", "text/html")
+			for _, value := range users {
+				fmt.Fprint(w, toStringFromUser(value)+"\n")
+			}
+		case "plain":
+			w.Header().Set("Content-Type", "text/plain")
+			for _, value := range users {
+				fmt.Fprint(w, toStringFromUser(value)+"\n")
+			}
+		// json
+		default:
+			res, err := json.Marshal(users)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			w.Header().Set("Content-Type", "application/json")
+			fmt.Fprint(w, string(res))
+		}
+	case "POST":
+		var user model.UserCreate
```

Comment:
これでもいいんですが、値を生成したあと内容を更新するのはメモリ的に非効率なのでこう書くことが慣習的に多いです。

```suggestion
		user := &model.UserCreate
```